thirdeye/thirdeye-pinot/src/main/java/com/linkedin/thirdeye/anomaly/detection/DetectionJobResource.java

```diff
@@ -249,4 +268,44 @@ public void run() {
 
     return Response.ok().build();
   }
+
+
+  @POST
+  @Path("/autotune/{id}")
+  public Response tuneAlertFilter(@PathParam("id") String id,
+      @QueryParam("startTime") Long startTime, @QueryParam("endTime") Long endTime)
+      throws Exception {
+    AnomalyFunctionDTO anomalyFunctionSpec = DAO_REGISTRY.getAnomalyFunctionDAO().findById(Long.valueOf(id));
+    AnomalyFunctionManager anomalyFunctionDAO = DAO_REGISTRY.getAnomalyFunctionDAO();
+    AlertFilter alertFilter;
+    Map<String, String> alertFilterParams = anomalyFunctionSpec.getAlertFilter();
+    if (alertFilterParams == null) {
+      alertFilter = new DummyAlertFilter();
+    } else if (alertFilterParams.get("type").toUpperCase().equals(ALPHA_BETA.toString())) {
```

Comment:
The alertFilterFactory should also take care of this case, i.e., we can get any type of filter from the factory regardless it comes from TE or anomaly library.